sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -89,7 +91,7 @@ AggregatorFactory createAggregatorFactory(String name, String fieldName, String
             return new DoubleFirstAggregatorFactory(name, fieldName, timeColumn);
           case STRING:
           case COMPLEX:
-            return new StringFirstAggregatorFactory(name, fieldName, timeColumn, maxStringBytes);
+            return new StringFirstAggregatorFactory(name, fieldName, timeColumn, null, maxStringBytes, true);
```

Comment:
is this right if we go forward with separate `foldColumn`? Shouldn't the complex case put `fieldName` in the `foldColumn`?

 If you drop `foldColumn` and use the selector factory + capabilities (my preference at least at the time of writing this comment) instead then this doesn't need to change for now I think. Though, this part will also need to change if the PR that adds ingest time support for numeric first/last, since we'll need to get the complex type name and pick the correct folding aggregator...